contracts/adapters/Onboarding.sol

```diff
@@ -310,62 +317,89 @@ contract OnboardingContract is
+        uint256 amount = proposal.amount;
         if (voteResult == IVoting.VotingState.PASS) {
             address tokenToMint = proposal.tokenToMint;
+            uint256 sharesRequested = proposal.sharesRequested;
             address applicant = proposal.applicant;
             BankExtension bank = BankExtension(dao.getExtensionAddress(BANK));
-            _mintTokensToMember(
-                dao,
-                tokenToMint,
-                applicant,
-                proposal.sharesRequested,
-                proposal.proposer,
-                proposal.token,
-                proposal.amount
-            );
-
-            address token = proposal.token;
-            if (token == ETH_TOKEN) {
-                bank.addToBalance{value: proposal.amount}(
-                    GUILD,
+            bool success =
+                _mintTokensToMember(
+                    dao,
+                    tokenToMint,
+                    applicant,
+                    sharesRequested,
+                    proposer,
                     token,
-                    proposal.amount
+                    amount
                 );
-            } else {
-                bank.addToBalance(GUILD, token, proposal.amount);
-
-                IERC20 erc20 = IERC20(token);
-                erc20.safeTransfer(address(bank), proposal.amount);
+            address daoAddress = address(dao);
+            if (success) {
+                if (token == ETH_TOKEN) {
+                    // On overflow failure return tokens to the proposer.
+                    try bank.addToBalance{value: amount}(GUILD, token, amount) {
+                        // do nothing
+                    } catch {
+                        _refundTribute(token, proposer, amount, "overflow:eth");
+                        // Remove the minted tokens
+                        bank.subtractFromBalance(
+                            applicant,
+                            tokenToMint,
+                            sharesRequested
+                        );
+                    }
+                } else {
+                    // On overflow failure return tokens to the proposer.
+                    try bank.addToBalance(GUILD, token, amount) {
+                        IERC20 erc20 = IERC20(token);
+                        erc20.safeTransfer(address(bank), amount);
+                    } catch {
+                        _refundTribute(
+                            token,
+                            proposer,
+                            amount,
+                            "overflow:erc20"
+                        );
+                        // Remove the minted tokens
+                        bank.subtractFromBalance(
+                            applicant,
+                            tokenToMint,
+                            sharesRequested
+                        );
+                    }
+                }
+
+                uint256 totalShares;
+                unchecked {
```

Comment:
is this still possible ? since we've limited internal tokens to uint88 I would imagine that there is no way we have an overflow no?